cmdx.py

```diff
@@ -3151,6 +3240,80 @@ def animated(self, recursive=True):
 
         return False
 
+    def findAnimatedPlug(self):
+        """Return the plug currently controlling the animation for this plug
+
+        Sometimes a plug isn't directly connected to an animCurve, for example
+        if the plug is constrained or in an animation layer. This method tracks
+        down where we should apply any new animation.
+
+        Example:
+            >>> _new()
+            >>> node = createNode("transform")
+            >>> node["tx"].findAnimatedPlug().plug() == node["tx"].plug()
+            True
+
+            # Constraint
+            >>> node["tx"] = {0:1} # Dummy keys so a pairBlend is created
+            >>> parent = createNode("transform")
+            >>> _ = cmds.parentConstraint(str(parent), str(node))
+            >>> blend = ls(type="pairBlend")[0]
+            >>> node["tx"].findAnimatedPlug().plug() == blend["inTranslateX1"].plug()
+            True
+
+            # Animation layers
+            >>> layer = encode(cmds.animLayer(at=node["tx"].path()))
+            >>> node["tx"].findAnimatedPlug().plug() == blend["inTranslateX1"].plug()
+            True
+            >>> cmds.animLayer(str(layer), e=True, preferred=True)
+            >>> animBlend = ls(type="animBlendNodeBase")[0]
+            >>> node["tx"].findAnimatedPlug().plug() == animBlend["inputB"].plug()
+            True
+
+            # Animation layer then constraint
+            >>> _new()
+            >>> node = createNode("transform")
+            >>> layer = encode(cmds.animLayer(at=node["tx"].path()))
+            >>> parent = createNode("transform")
+            >>> _ = cmds.parentConstraint(str(parent), str(node))
+            >>> animBlend = ls(type="animBlendNodeBase")[0]
+            >>> node["tx"].findAnimatedPlug().plug() == animBlend["inputA"].plug()
```

Comment:
I think this better be `is` rather than `==`, as `==` would not compare the plugs but the plug values.

Reply:
I tried changing it to `is` and it didn't work for some reason. Shouldn't `==` work here as I'm using the `.plug()` method to grab the underlying `MPlug`?

Reply:
Hm, you are right. Initially, plugs were reused such that their physical Python instance was stored in a global dictionary, similar to nodes. But, I think that got removed in favour of stability.. Easy to get some nasty crashes when accessing a plug that isn't valid anymore.

So `is` probably isn't going to work.. I tried finding whether there was another way to spot their equivalence, and there is `__hash__` to enable storage of plugs in a dictionary. They just look at `plug.path()` which is fair; if their paths are the same, then we can trust they are the same. So I would check against `.path()` instead.